python/gstaichi/lang/_fast_caching/src_hasher.py

```diff
@@ -0,0 +1,75 @@
+from typing import Any, Iterable, Sequence
+
+from pydantic import BaseModel
+
+from . import args_hasher, config_hasher, function_hasher
+from .fast_caching_types import FunctionSourceInfo, HashedFunctionSourceInfo
+from .hash_utils import hash_string
+from .pyside_cache import PysideCache
+
+
+def create_cache_key(kernel_source_info: FunctionSourceInfo, args: Sequence[Any]) -> str | None:
+    """
+    cache key takes into account:
+    - arg types
+    - cache value arg values
+    - kernel function (but not sub functions)
+    - compilation config (which includes arch, and debug)
+    """
+    args_hash = args_hasher.hash_args(args)
+    if args_hash is None:
+        return None
+    kernel_hash = function_hasher.hash_kernel(kernel_source_info)
+    config_hash = config_hasher.hash_compile_config()
+    cache_key = hash_string(f"{kernel_hash}_{args_hash}_{config_hash}")
+    return cache_key
+
+
+class CacheValue(BaseModel):
+    hashed_function_source_infos: list[HashedFunctionSourceInfo]
+
+
+def store(cache_key: str, function_source_infos: Iterable[FunctionSourceInfo]) -> None:
+    """
+    Note that unlike other caches, this cache is not going to store the actual value we want. This cache is only used for verification
```

Comment:
It should start with a one-line short description before details.

I would advise you not to violate linewidth policy in docstring, even if you formatter allows this.